### PR TITLE
E1602 - Refactor different question types from quiz feature

### PR DESCRIPTION
**Pull request by WumpaFruit**
We have refactored the four methods `edit`, `view_question_text`, `complete`, and `view_completed_question` of the three different types of questions. We have also created unit tests for each of these methods.

### CODE REVIEW

File: app/views/grades/_scripts.html.erb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<SCRIPT>
```
Comment (yangsong8) [inline app/views/grades/_scripts.html.erb]: Is this file added by you guys?

File: app/views/grades/_reviews.html.erb [hunk 2]
```diff
@@ -30,59 +30,73 @@
     <!--And if one reviewer did not do first-round review and did second round directly, Expertiza will display only second round.-->
```
Comment (yangsong8) [inline app/views/grades/_reviews.html.erb]: I got confused, this file is not changed by you guys, right?

**Comment by yangsong8** [comment]
Hi guys, it seems that you have not forked the master branch correctly before you do the changes. It seems your pull request, if merged, will revert some of the change we just did to master branch. I suggest you back up you changes, fork the master branch again, make your changes, then submit another pull request. Thanks.

**Comment by WumpaFruit** [comment]
Darshan or Tanay, can one of you take care of this? My midterm is today.

**Comment by yangsong8** [comment]
Hi, is this the old pull request? If yes, could you close it?

**Comment by WumpaFruit** [comment]
Closed. Please refer to https://github.com/expertiza/expertiza/pull/663 for a fixed pull request.

**Comment by WumpaFruit** [comment]
I just closed it.

On Wed, Mar 30, 2016 at 1:13 PM, Yang Song <notifications@github.com> wrote:

> Hi, is this the old pull request? If yes, could you close it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/expertiza/expertiza/pull/660#issuecomment-203534032>
>



-- 
*Joseph Sankar (About Me) <http://josephsankar.me>*
Master of Computer Science, Class of 2016, North Carolina State University
Bachelor's in Computer Science, Class of 2015, North Carolina State
University

**Pull request closed without merging**